app/hooks/use-socket-event.ts

```diff
@@ -1,11 +1,12 @@
 import { useEffect, useState } from "react";
-import { socket } from "@/socket";
+import { useSocket } from "@/components/socket-provider";
 import type { ServerToClientEvents } from "@/types";
 
 export function useSocketEvent<K extends keyof ServerToClientEvents>(event: K) {
   const [result, setResult] = useState<
     Parameters<ServerToClientEvents[K]> | []
   >([]);
+  const { socket } = useSocket();
```

Comment:
fail fast (provided you initialize context to `null`)

```suggestion
  const { socket } = useSocket();
  const [result, setResult] = useState<
    Parameters<ServerToClientEvents[K]> | []
  >([]);
```